src/mlpack/methods/ann/regularizer/orthogonal_regularizer_impl.hpp

```diff
@@ -0,0 +1,65 @@
+/**
+ * @file orthogonal_regularizer_impl.hpp
+ * @author Saksham Bansal
+ *
+ * Implementation of OrthogonalRegularizer class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_ORTHOGONAL_REGULARIZER_IMPL_HPP
+#define MLPACK_METHODS_ANN_ORTHOGONAL_REGULARIZER_IMPL_HPP
+
+// In case it hasn't been included.
+#include "orthogonal_regularizer.hpp"
+
+namespace mlpack {
+namespace ann {
+
+OrthogonalRegularizer::OrthogonalRegularizer(double factor) :
+    factor(factor)
+{
+  // Nothing to do here.
+}
+
+template<typename MatType>
+void OrthogonalRegularizer::Evaluate(const MatType& weight, MatType& gradient)
+{
+  arma::mat grad = arma::mat(arma::size(weight), arma::fill::zeros);
```

Comment:
This is just fine, just wanted to point out there is a function to shorten this:

```c++
arma::mat grad = arma::zeros(arma::size(weight));
```